packages/dashboard-frontend/src/components/UntrustedSourceModal/index.tsx

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+
+import SessionStorageService, { SessionStorageKey } from '@/services/session-storage';
+
+export type Props = {
+  location: string;
+  isOpen: boolean;
+  onClose?: () => void;
+  onContinue: () => void;
+};
+export type State = {
+  isTrusted: boolean;
+  trustAllCheckbox: boolean;
+};
+
+export class UntrustedSourceModal extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      isTrusted: UntrustedSourceModal.isSourceTrusted(props.location),
+      trustAllCheckbox: false,
+    };
+  }
+
+  static isSourceTrusted(location: string): boolean {
+    const trustedSources = SessionStorageService.get(SessionStorageKey.TRUSTED_SOURCES);
+    if (!trustedSources) {
+      return false;
+    } else if (trustedSources === 'all') {
+      return true;
+    }
+
+    const trustedSourcesArray = trustedSources.split(',');
+    return trustedSourcesArray.includes(location);
+  }
+
+  static updateTrustedSources(location: string, trustAll: boolean): void {
+    const trustedSources = SessionStorageService.get(SessionStorageKey.TRUSTED_SOURCES);
+    if (trustedSources === 'all') {
+      return;
+    } else if (trustAll) {
+      SessionStorageService.update(SessionStorageKey.TRUSTED_SOURCES, 'all');
+    } else {
+      const prevArray = trustedSources ? trustedSources.split(',') : [];
+      const trustedSourcesSet = new Set(prevArray);
+      trustedSourcesSet.add(location);
+      const nextArray = Array.from(trustedSourcesSet);
+      SessionStorageService.update(SessionStorageKey.TRUSTED_SOURCES, nextArray.join(','));
+    }
+  }
+
+  private handleTrustAllToggle(checked: boolean): void {
+    this.setState({ trustAllCheckbox: checked });
+  }
+
+  private handleClose(): void {
+    this.setState({ trustAllCheckbox: false });
+
+    this.props.onClose?.();
+  }
+
+  private handleContinue(): void {
+    UntrustedSourceModal.updateTrustedSources(this.props.location, this.state.trustAllCheckbox);
+
+    this.setState({ trustAllCheckbox: false });
+
+    this.props.onContinue();
+    this.props.onClose?.();
+  }
+
+  private buildModalFooter(): React.ReactNode {
+    return (
+      <React.Fragment>
+        <Button
+          key="continue"
+          variant={ButtonVariant.primary}
+          onClick={() => this.handleContinue()}
+        >
+          Continue
+        </Button>
+        <Button key="cancel" variant={ButtonVariant.link} onClick={() => this.handleClose()}>
+          Cancel
+        </Button>
+      </React.Fragment>
+    );
+  }
+
+  private buildModalBody(): React.ReactNode {
+    const { trustAllCheckbox: isChecked } = this.state;
+    return (
+      <TextContent>
+        <Text>
+          Click <b>Continue</b> to proceed creating a new workspace from this source.
+        </Text>
+        <Checkbox
+          id="trust-all-repos-checkbox"
+          isChecked={isChecked}
+          label="Do not ask me again (within this session)"
```

Comment:
what does `(within this session)` mean exactly? I though we are going to store preferences within k8s cm, no?

Reply:
It means the browser page session. You're right, I missed implementing storing this flag in the configmap, going to fix that.

Reply:
I think we need to remove `(within this session)` to avoid confusion